sql/analyzer/indexes.go

```diff
@@ -760,10 +779,19 @@ func getMultiColumnIndexForExpressions(
 	if lookup == nil {
 		return nil, nil
 	}
+	var lookupExpr sql.Expression
+	for _, m := range matchedExprs {
+		if lookupExpr == nil {
+			lookupExpr = m.comparison
+		} else {
+			lookupExpr = expression.NewAnd(lookupExpr, m.comparison)
```

Comment:
This is what I'm talking about. I think you're constructing the lookup expression in such a way that it doesn't match the filter's predicate in pushdown, so it can't be eliminated.

Reply:
The rearrangements concern me but the only eliminations I miss are inversions. I'll take another look but I am pretty sure this is immaterial to the filter elimination.

Reply:
Is there a different analyzer rule that does anything with filters? I've spent an hour looking at this and I still think `lookupExpr` is distinct from the Filter expression. We never set a Filter expr as the lookup expr, if anything we do the opposite.

Reply:
the problem turned out to be a missing conjugate split, unrelated to this